### PR TITLE
Weekly report

### PR DESCRIPTION
**Pull request by morfi11**
need your help with the date-range component and making it weekly with the @Input() rangeWeekly parameter.

### CODE REVIEW

File: src/app/date-range/date-range.component.ts [hunk 1]
```diff
@@ -63,13 +62,19 @@ export class DateRangeComponent implements OnInit {
   constructor(public settings: ApplicationSettings) {
     let today = new Date();
 
-    this.fromDate.value = new Date(today.getFullYear(), today.getMonth(), 1);
-    this.toDate.value = this.getEndOfMonth();
+    if (this.rangeWeekly) {
```
Comment (noam-honig) [inline src/app/date-range/date-range.component.ts:65]: you probably want to put this in the ngOnInit - at which point the input values will be set

File: src/app/families/DeliveryStatus.ts [hunk 2]
```diff
@@ -85,6 +85,10 @@ export class DeliveryStatusColumn extends ValueListColumn<DeliveryStatus> {
   isProblem() {
     return this.isGreaterOrEqualTo(DeliveryStatus.FailedBadAddress).and(this.isLessOrEqualTo(DeliveryStatus.FailedOther));
   }
+  isNotProblem() {
+    return this.isLessOrEqualTo(DeliveryStatus.SuccessLeftThere).and(this.isDifferentFrom(DeliveryStatus.Frozen));
```
Comment (noam-honig) [inline src/app/families/DeliveryStatus.ts:89]: this will include also ready for delivery and self pickup - I hope that's what you wanted

**Review by noam-honig — CHANGES_REQUESTED** [review]

File: src/app/families/DeliveryStatus.ts (hunk 2 above)
Comment (morfi11) [reply src/app/families/DeliveryStatus.ts:89]: yes. exactly what I wanted. everything that either received or still on the way.

File: src/app/date-range/date-range.component.ts (hunk 1 above)
Comment (morfi11) [reply src/app/date-range/date-range.component.ts:65]: so the input value is only available there?

Comment (morfi11) [reply src/app/date-range/date-range.component.ts:65]: yes.... I just realized it was the constructor. Thank you :) it works now

**Comment by morfi11** [comment]
@noam-honig - ready for your review and merge

**Pull request merged**